akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala

```diff
@@ -159,7 +159,13 @@ object Sink {
    * [[Attributes]] of the [[Sink]] returned by this method.
    */
   def fromMaterializer[T, M](factory: (Materializer, Attributes) => Sink[T, M]): Sink[T, Future[M]] =
-    Sink.fromGraph(new SetupSinkStage(factory))
+    Flow
+      .fromMaterializer({ (mat, attr) =>
+        Flow.fromGraph(GraphDSL.create(factory(mat, attr)) { b => sink =>
+          FlowShape(sink.in, b.materializedValue.outlet)
+        })
+      })
+      .to(Sink.head)
```

Comment:
@johanandren I think this was the most straightforward way to fix this; hope that it's ok.

Reply:
Clever. Not entirely sure of the performance implications, graphdsl now vs a submaterialization before, but perhaps that's ok.